frame/staking/src/lib.rs

```diff
@@ -960,6 +960,56 @@ pub mod pallet {
 
 		/// Weight information for extrinsics in this pallet.
 		type WeightInfo: WeightInfo;
+
+		/// The list of thresholds separating the various voter bags.
+		///
+		/// Voters are separated into unsorted bags according to their vote weight. This specifies
+		/// the thresholds separating the bags. A voter's bag is the largest bag for which the
+		/// voter's weight is less than or equal to its upper threshold.
+		///
+		/// When voters are iterated, higher bags are iterated completely before lower bags. This
+		/// means that iteration is _semi-sorted_: voters of higher weight tend to come before
+		/// voters of lower weight, but peer voters within a particular bag are sorted in insertion
+		/// order.
+		///
+		/// # Expressing the constant
+		///
+		/// This constant must be sorted in strictly increasing order. Duplicate items are not
+		/// permitted.
+		///
+		/// There is an implied upper limit of `VoteWeight::MAX`; that value does not need to be
+		/// specified within the bag. For any two threshold lists, if one ends with
+		/// `VoteWeight::MAX`, the other one does not, and they are otherwise equal, the two lists
+		/// will behave identically.
+		///
+		/// # Calculation
+		///
+		/// It is recommended to generate the set of thresholds in a geometric series, such that
+		/// there exists some constant ratio such that `threshold[k + 1] == (threshold[k] *
+		/// constant_ratio).max(threshold[k] + 1)` for all `k`.
+		///
+		/// The helpers in the `voter_bags::make_bags` module can simplify this calculation. To use
+		/// them, the `make-bags` feature must be enabled.
+		///
+		/// # Examples
+		///
+		/// - If `VoterBagThresholds::get().is_empty()`, then all voters are put into the same bag,
+		///   and iteration is strictly in insertion order.
+		/// - If `VoterBagThresholds::get().len() == 64`, and the thresholds are determined
+		///   according to the procedure given above, then the constant ratio is equal to 2.
```

Comment:
why is this the case? I think there is a missing maximum size of the thresholds that we're making this deduction based of?

Reply:
So, that comment is very slightly inaccurate: it's actually the case that the constant ratio is 2 when there are 65 thresholds. 

If you think of it from the other way around, if you wanted to split the `u64` space into geometric thresholds as simply as possible, you'd make the thresholds list be `(0..64).map(|n| 1 << n).chain(std::iter::once(u64::MAX))`. Of course, that ends up being 65 items long. That also assumes that the existential deposit is 1. 

We demo that in the list of test thresholds.